### PR TITLE
Add functionality for users to retrieve template by Name as well as UUID

### PR DESCRIPTION
**Pull request by Belchy06**
Signed-off-by: William Belcher <william.belcher@hotmail.com>

## Description

This PR adds the ability for Tink to retrieve workflows by either UUID or by Name. If the value after `tink template get` is not a valid UUID, Tink will default to searching for the value in the Name field of the templates. 

This PR also adds a TemplateClient to the Tink client, allowing other services such as Boots to retrieve templates.

## Why is this needed

This feature is required for the project I am working on but I am sure there are many other use cases I'm unaware of. Also, https://github.com/tinkerbell/boots/issues/178

## How Has This Been Tested?
This has been tested in the VM's based on the [Local Setup with Vagrant](https://docs.tinkerbell.org/setup/local-vagrant/) tutorial, as well as in our personal deployment

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

**Comment by Belchy06** [comment]
@tstromberg I had done something funky with my last PR for this and was unable to get the DCO signed off correctly.

This PR is essentially the same as the last, just compressed into a single commit with DCO signed off.

File: client/main.go [hunk 1]
```diff
@@ -161,3 +161,12 @@ func TinkWorkflowClient() (workflow.WorkflowServiceClient, error) {
...
 }
+
+// TinkTemplateClient creates a new template client
+func TinkTemplateClient() (template.TemplateServiceClient, error) {
+	conn, err := GetConnection()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return template.NewTemplateServiceClient(conn), nil
+}
```
Comment (mmlb) [inline client/main.go:172]: Lets get rid of this from this PR please. boots does not need template access so lets wait until an actual need comes up for this.

File: protos/template/template.pb.go [hunk 2]
```diff
@@ -27,6 +24,8 @@ import (
 	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
 	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
 	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+	reflect "reflect"
+	sync "sync"
```
Comment (mmlb) [inline protos/template/template.pb.go:28]: please drop this change.

**Review by mmlb — CHANGES_REQUESTED** [review]

**Comment by Belchy06** [comment]
Sorry I haven't gotten around to those changes yet @tstromberg , I've been super busy. Will get around to them this weekend.

**Comment by tstromberg** [comment]
@Belchy06 - hey, just checking in to see if there is anything you would like help with here. I'm eager to see this PR be merged so that folks can use it :)

**Pull request closed without merging**